### PR TITLE
Discord Webhooks endpoint 

### LINKED ISSUE #5: [Feature request] Send announcement to Discord server upon event creation using Webhooks
Refer to https://github.com/SVijayB/OSC-DBot/issues/20#issuecomment-988568136 for more details.

### PR DESCRIPTION
**Pull request by officialakhil**
### Description
- This PR adds a POST `/api/event/announcement?api_key=` endpoint which takes `{ "event_id": 1 }` as request data and sends an announcement in the discord server. 

### Related Issue

This PR closes #5 

### Motivation and Context

Refer issue #5 

### Screenshots (if any)
image
image

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Review by dat-adi — APPROVED** [review]
Changes look pretty good.
- [x] Environment variables addition.
- [x] Webhook to work with the Discord Bot.

Testing has yet to be done, but, PR description suffices.
Approved on my side.

File: src/utils/discord_webhook.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        description=event["eventDescription"],
+        color=0x2F3136,
+        timestamp="now",
+    )
+
+    embed.set_author(
+        name="Vijay",
+        url="https://github.com/SVijayB",
+        icon_url="https://avatars.githubusercontent.com/svijayb",
+    )
```
Comment (dat-adi) [inline src/utils/discord_webhook.py:19]: This will need to be converted to OSC's GitHub profile.

**Review by dat-adi — CHANGES_REQUESTED** [review]
## Changes required
- [x] Author for the embed needs to be the Open Source Community.

File: src/utils/discord_webhook.py (hunk 1 above)
Comment (dat-adi) [reply src/utils/discord_webhook.py:19]: Changes to be made.
```
name => Open Source Community: VIT-AP
url => https://github.com/Open-Source-Community-VIT-AP
icon_url => https://avatars.githubusercontent.com/open-source-community-vit-ap
```

File: src/routes/events/event.py [hunk 2]
```diff
@@ -32,3 +35,41 @@ def latest_event():
...
+        if event["id"] == event_id:
+            required_event = event
+            break
+
+    if not required_event:
+        return f"No event found with ID: {event_id}", 404
+
+    send_discord_announcement(WEBHOOK_URL, required_event)
+
+    return jsonify(required_event)
```
Comment (SVijayB) [inline src/routes/events/event.py:75]: Return success/failed message instead of data.

**Review by SVijayB — CHANGES_REQUESTED** [review]
Could you also reformat the code using black code formatter?

File: src/routes/events/event.py [hunk 3]
```diff
@@ -32,3 +35,41 @@ def latest_event():
...
+    if not WEBHOOK_URL:
+        return (
+            "Internal Error: No Webhook URL is configured in environment variables",
+            500,
+        )
+
+    data = connection()
+    post_data = request.json
+
+    if not post_data or not post_data.get("event_id"):
```
Comment (SVijayB) [inline src/routes/events/event.py:59]: If no event ID was provided in POST request, then take event_id from the latest event.
Why? Event ID data is auto-generated by Django and added to the database. When a new event is created, this endpoint is called without an event id.

**Review by SVijayB — CHANGES_REQUESTED** [review]

File: src/utils/discord_webhook.py (hunk 1 above)
Comment (officialakhil) [reply src/utils/discord_webhook.py]: Handled

File: src/routes/events/event.py (hunk 2 above)
Comment (officialakhil) [reply src/routes/events/event.py]: Handled in cd6301f

File: src/routes/events/event.py (hunk 3 above)
Comment (officialakhil) [reply src/routes/events/event.py:60]: Handled in 3524cc2

Comment (officialakhil) [reply src/routes/events/event.py:60]: image

**Comment by officialakhil** [comment]
> Could you also reformat the code using black code formatter?

image
Looks like I already have it applied. 

@SVijayB @dat-adi -- handled all the requested changes :v:

**Review by SVijayB — APPROVED** [review]
Changes have been made as requested. 
Approved PR for merge.
Good work @officialakhil!

**Pull request merged**